crates/ark/src/viewer.rs

```diff
@@ -41,17 +51,51 @@ fn emit_html_output(iopub_tx: Sender<IOPubMessage>, path: String) -> Result<()>
 }
 
 #[harp::register]
-pub unsafe extern "C" fn ps_html_viewer(url: SEXP) -> anyhow::Result<SEXP> {
+pub unsafe extern "C" fn ps_html_viewer(
+    url: SEXP,
+    label: SEXP,
+    height: SEXP,
+    is_plot: SEXP,
+) -> anyhow::Result<SEXP> {
     // Convert url to a string; note that we are only passed URLs that
     // correspond to files in the temporary directory.
     let path = RObject::view(url).to::<String>();
+    let label = match RObject::view(label).to::<String>() {
+        Ok(label) => label,
+        Err(_) => String::from("R"),
+    };
     match path {
         Ok(path) => {
-            // Emit the HTML output
+            // Emit HTML output
             let main = RMain::get();
             let iopub_tx = main.get_iopub_tx().clone();
-            if let Err(err) = emit_html_output(iopub_tx, path) {
-                log::error!("Failed to emit HTML output: {:?}", err);
+            if main.session_mode == SessionMode::Notebook {
+                // In notebook mode, send the output as a Jupyter display_data message
+                if let Err(err) = emit_html_output_jupyter(iopub_tx, path, label) {
+                    log::error!("Failed to emit HTML output: {:?}", err);
+                }
+            } else {
+                let is_plot = RObject::view(is_plot).to::<bool>();
+                let height = RObject::view(height).to::<i32>();
+                let params = ShowHtmlFileParams {
+                    path,
+                    title: label.clone(),
+                    height: match height {
+                        Ok(height) => height.into(),
+                        Err(_) => 0,
+                    },
```

Comment:
Would you mind lifting this error checking out into its own line with a `log::warn()` in the failure path? 

Something like

```rust
let height = match height {
  Ok(height) => height.into(),
  Err(err) => {
    log::warn!("Can't convert `height` into an i32, using `0` as a fallback: {err:?}");
    0
  },
}
```

Same for `is_plot`.

That possibly could have helped catch the `NULL` `height` i mentioned earlier

Reply:
can do!